src/coord/src/catalog/builtin.rs

```diff
@@ -1227,6 +1227,7 @@ pub const PG_ATTRIBUTE: BuiltinView = BuiltinView {
     mz_objects.oid as attrelid,
     mz_columns.name as attname,
     mz_types.oid AS atttypid,
+    0::pg_catalog.int2 AS attlen,
```

Comment:
The [PG docs](https://www.postgresql.org/docs/13/catalog-pg-attribute.html) say:
> A copy of pg_type.typlen of this column's type

So I think if this exists, then `pg_type.typlen` should probably also exist.

Also don't think this _needs_ to be zero; we have a function that expresses the number of bytes each scalartype consumes at `src/pgrepr/src/types.rs`, `typlen`. Requires some dancing around with the types but is maybe tractable? Looks like we'd need to embed something in `TypeInner::Base` to express what it was derived from.

If we go with zero, would like a `TODO` or a link to an issue mentioning this is a quick fix for pgcompat, but deserves more TLC at some point.

Reply:
I've rejiggered attlen so it pulls from `pg_type.typlen`, but typelen i've set to null. i looked into plumbing down the `typlen()` function but it involved more stuff than i wanted in this pr. can come later.

regarding adding a todo comment: our docs already say

> Most are lacking some columns that are present in PostgreSQL, or if they do include the column the result set its value may always be NULL.

and we have tons of other NULL things that we know should be improved. i don't think this one deserves special callout. ok with you?

Reply:
Ah, you're right and I hadn't looked at the user docs. My oversight and this should go in as the others have.